hazelcast/src/test/java/com/hazelcast/client/map/impl/querycache/ClientQueryCacheRecreationTest.java

```diff
@@ -159,4 +162,8 @@ public void entryAdded(EntryEvent event) {
         assertTrueAllTheTime(assertTask, 3);
     }
 
+    private static void assertCacheSizeEventually(QueryCache cache, int expectedCacheSize) {
+        assertTrueEventually(() -> assertEquals(expectedCacheSize, cache.size()));
```

Comment:
why is this different from https://github.com/hazelcast/hazelcast/pull/18233/files#diff-583e78ff420d5c3c2799f5741af24ac5bb6f21b11effd46d09013e614c6d926fL112 in terms of the logic?

Reply:
`assertTrueEventually` below has additional recovery logic in the catch block. 
New method is just plain check.

Reply:
That is right and exactly why I was questioning if we do not need the same check at this step of the test. do we?

Reply:
I might be wrong but from what I understand, tryRecover removes some inconsistencies between the operations `recreate` and additional incoming events. Since at this point, we did not call `recreate`, therefore I thought the extra tryRecovery would be unnecessary. However, I think using the same construct will not cause any trouble. If it is OK for you, I can change this static function to use a `tryRecovery` call.

Reply:
i would choose to be consistent.